File: day_01/python/part2.py
input = open('../input.txt').read()

floor = 0

for i in range(len(input)):
    if input[i] == '(':
        floor += 1
    else:
        floor -= 1

    if floor < 0:
        print('Entered the basement on position #{}'.format(i+1))
        break

File: day_01/js/part2.js
const fs = require('fs');

const input = fs.readFileSync('../input.txt', 'utf8');

let index = 0;
let floor = 0;

while (floor >= 0) {
    floor += input.charAt(index) === '('
        ? 1
        : -1;

    index++;
}

console.log(index);
